### PR TITLE
[MO] - [system test] -> oauth authorization superuser breaks rules

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: see-quick <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR:
1) adding test case for oauth authorization with combination of `superusers`, which breaks all rules defined in the `authorization server`. 
2) changing communication inside `OauthAuthorizationST` to plain for simplicity.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ void verifyVolumeNamesAndLabels(int kafkaReplicas, int diskCountPerReplica, int
     }
 
     @Test
-    void testPersistentStorageSize() throws Exception {
+    void testPersistentStorageSize() {
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: It would make reviews of ST PRs lot easier if every PR didn't contain bunch of removals of `throws Exception`. Is this in any way related to the actual PR or is it just something you noticed while doing the PR? If it is related to the PR it is of course ok. But if not, please open one PR to fix all of these to make the reviews easier.

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java [hunk 2]
```diff
@@ -61,11 +61,11 @@
     @Description("As a member of team A, I should be able to read and write to all topics starting with a-")
     @Test
     void smokeTestForClients() throws IOException, InterruptedException, ExecutionException, TimeoutException {
-        Future<Integer> producer = teamAOauthKafkaClient.sendMessagesTls(TOPIC_A, NAMESPACE, CLUSTER_NAME, TEAM_A_CLIENT,
-                MESSAGE_COUNT, "SSL");
+        Future<Integer> producer = teamAOauthKafkaClient.sendMessages(TOPIC_A, NAMESPACE, CLUSTER_NAME,
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: What is the reason to move so massively to not using TLS? Again, it looks a bit unrelated to the main PR.

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java [hunk 3]
```diff
@@ -161,15 +159,81 @@ void testTeamAWriteToTopicStartingWithXAndTeamBReadFromTopicStartingWithX() thro
...
+
+        Future<Integer> consumer = teamBOauthKafkaClient.receiveMessages(topicName, NAMESPACE, CLUSTER_NAME,
+                MESSAGE_COUNT, "x_consumer_group_b");
+
+        assertThat(consumer.get(Constants.GLOBAL_CLIENTS_TIMEOUT, TimeUnit.MILLISECONDS), is(MESSAGE_COUNT));
+    }
+
+    @Description("As a superuser of team A and team B, i am able to break defined authorization rules")
+    @Test
+    void testSuperUserWithOauthAuthorization() throws InterruptedException, ExecutionException, TimeoutException, IOException {
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: I think this tests the super users well. But this test it self doesn't contain any verifications that the authorization alone works. Do you think it is sufficient that it is coverd in other tests? Or should we have something like:
1. Setup the clusetr without super users
2. Test that they cannot read and write
3 Modify the cluster to add super users
4. Test that they can now read and write.

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java (hunk 1 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Yeah, sorry for that...

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: I realized that it would be better to test just a "authorization" and let the encryption be.

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java (hunk 3 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: I can add another test case for clarity with that scenario.

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java (hunk 2 above)
Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: Ok, makes sense.

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java (hunk 3 above)
Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: I think it would be good, because right now here you do not really know whether the authorization works at all.

Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java]: I have decided to make it really simple and did these scenario in one test. It will be really time consuming if i do two separate test cases...

**Comment by scholzj** [comment]
@ppatierno Do you think you can get to review this? Or should we ask Tom?
